enhancements/network/egressip_for_ovn_kubernetes.md

```diff
@@ -0,0 +1,85 @@
+---
+title: host-level-openvswitch
+authors:
+  - "@alexanderConstantinescu"
+reviewers:
+  - "@danwinship"
+approvers:
+  - TBD
+creation-date: 2020-06-05
+last-updated: 2019-06-05
+status: provisional
+
+---
+
+# EgressIP for OVN-Kubernetes
+
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+EgressIP is currently a feature for openshift-sdn and is planned to be implemented for OVN-Kubernetes and subsequently integrated in the 4.6 release of OpenShift. This feature implementation for OVN-Kubernetes requires feature parity with openshift-sdn, but also aims for a feature improvement which is detailed in this document. 
+
+### Background
+
+EgressIP, as it's currently implemented for openshift-sdn, allows traffic exiting a cluster namespace to have a fixed IP address. This IP is attached to the primary network interface on one of the cluster nodes. The IP address can be manually assigned or left up to the network plugin to automatically assign. If automatic assignment is done, the egress IP is then considered to be highly available, meaning: if the node currently hosting that IP falls off the cluster, the network plugin re-assigns a new egress IP to another cluster node as a fallback mechanism.
+
+This enhancement proposal focuses on adding a `podSelector` to the `EgressIP` object, allowing a user to specify a subset of pods within the namespace which should have an egress IP defined.  
+
+## Motivation
+
+The motivation behind this requirement comes from product management, and essentially boils down to allowing cluster admins the freedom on a lower granular level to control cluster egress traffic.  
+
+### Goals & Requirements
+
+1. The egress IP functionality for OVN-Kubernetes needs to have feature parity with openshift-sdn. 
+2. The egress IP functionality for OVN-Kubernetes should allow pod selection within a namespace when defining the IP of egress traffic. 
+3. The egress IP functionality for OVN-Kubernetes should support single-stack IPv4/IPv6 as well as dual-stack, in line with global OpenShift requirements for all networking functionality.
+ 
+### Constraints
+
+1. No pod can match multiple `EgressIP` objects, leading to multiple egress IP assignments at the same time. 
```

Comment:
I think this can _theoretically_ be possible, but do we want such a "feature"/"trap" @danwinship ?